### ISSUE TITLE
Continually increasing memory use

### ISSUE THREAD

**Issue by gnw-gomes** [Fluorescence-Tools/avtraj#4]
I've written a loop to write the FRET efficiency calculated using avtraj for each frame of a long MD simulation to a text file, however the memory use continues to increase until I'm using 100% of my computer memory.

I thought maybe its because python is inappropriately buffering the contents to be written to the file, but flushing the output to the file doesn't help either. I suspect that I'm maybe not using avtraj or mdtraj in the most efficient way? 

**Expected behaviour:** memory use should not grow with each loop iteration
**Observed behaviour:** memory use grows with each loop iteration

```
with open(filename,'w',newline='\n') as f: 
    f.write('#Label Avtraj Al488Al647 Ro=%f\n' %(R0))
    for i in range(len(av_traj_acceptor)):
        E=av_traj_acceptor[i].mean_fret_efficiency(av_traj_donor[i],forster_radius=R0)
        f.write('%f \n' % (E))
        #Write to file NOW
        f.flush()
        os.fsync(f) 
```

**Comment by gnw-gomes** [comment]
I found a solution which gets around this - but I'm still curious if or what I was doing wrong before.

Previously I loaded all 17 dcd files with mdtraj.load() and then did the calculations and file writing loop.

Now, I load one dcd file at a time, calculate the FRET efficiencies, perform file writing loop, and then move to next dcd file. However, I'm still not clear why the previous loop would use more memory as it proceeds.

**Comment by tpeulen** [comment]
I can imagine two options that can result in this behavior. 

1. The the python garbage collector (gc) is a bit too lazy, i.e., the gc only collects stray python (without objects without references) not very often. This could result in an increased memory usuage in a loop. 
2. Trajectories are referenced and thus not collected.

Using the garbage collector module (``import gc``) you can force the GC to collect ``gc.collect()`` unreferenced objects to clean up the memory. This way you can test if option 1 is causing your memory issues and the python gc is to blame. (GC comes at a performance cost. Thus python invokes the GC only very rarely.)

**Comment by tpeulen** [comment]
Sorry, I needed to check the source code. The behavior you describe is in fact expected. The library caches the computes dye distributions for frames that have been already processed.

Of course for cases where large trajectories are processes the memory usage will increase continously. I will modify the default behavior. In future computations will only be cached if this is explicilty demanded.

**Comment by tpeulen** [comment]
This is fixed by 1ef4379
